File: Memento/Java/Pojo/Caretaker.java
package Pojo;

import Memento.*;

public class Caretaker {
    private Memento memento;
    // ---------------------------
     public void setMemento(Memento memento) {
        this.memento = memento;
    }
    // ---------------------------
     public Memento getMemento() {
        return this.memento;
    }
}

File: Memento/cpp/Caretaker.cpp
#include <bits/stdc++.h>
#include "Originator.cpp"

using namespace std;

class Caretaker{
    private:
        Memento* memento;
    public:
        void setMemento(Memento*);
        Memento* getMemento();
};

void Caretaker::setMemento(Memento* memento){
    this->memento=memento;
}

Memento* Caretaker::getMemento(){
    return this->memento;
}
